### ISSUE TITLE
Toggle the visibility of navigation menu items

### ISSUE THREAD

**Issue by jeremystretch** [netbox-community/netbox#9070]
### NetBox version

v3.2.0

### Feature type

New functionality

### Proposed functionality

Enable NetBox administrators to disable certain navigation menu headings (e.g. Wireless or Circuits). This would most likely be implemented as a dynamic configuration option.

### Use case

This has come up a few times in various forums. I think the main theme is wanting to "hide" areas of NetBox that are unused. With the change to programmatically-rendered menu items in v3.0, I think this is more feasible than it has been in the past.

**Comment by stephanblanke** [comment]
I would propose to tie the visibility of menu items to the users permissions, instead of a separate option in the admin panel. Usually we want to hide an object type for the user in the web interface and the API.

**Comment by emersonfelipesp** [comment]
> I would propose to tie the visibility of menu items to the users permissions, instead of a separate option in the admin panel. Usually we want to hide an object type for the user in the web interface and the API.

I agree with @stephanblanke, having a global way of manage which navigation buttons must appear is cool. But for me, the real solution is to have this related to the permission group.

I work at an ISP and there are Netbox users which only uses the "Devices (DCIM)" section while others uses only the logical ones, like "Virtualization" and "IPAM".

But again, having a global visibility configuration is fine too, as none of the users of Netbox at the ISP uses the "Wireless" section.

**Comment by joaolucasmacedo** [comment]
That is pretty nice, but I would go further and consider some configuration to hide certain fields from objects, because there is so much information but mos people don't use all of them. So, if I'm proposing something feasible, it would be nice to have a way to hide some fields making navigation and object manipulation much cleaner and quicker.

**Comment by emersonfelipesp** [comment]
> That is pretty nice, but I would go further and consider some configuration to hide certain fields from objects, because there is so much information but mos people don't use all of them. So, if I'm proposing something feasible, it would be nice to have a way to hide some fields making navigation and object manipulation much cleaner and quicker.

Maybe instead of "simply" manage the Model permissions as Netbox currently do, it would be necessary to create some mechanism to manage the field permissions of the Model.

**Comment by lknflexfone** [comment]
we can defently concour on this feature, would be nice if the view was bound the permissions model so you only presented what your permissions gave you access to.

**Comment by jeremystretch** [comment]
I'm going to tag this for v3.3 with the modification that permissions will be used to control the display of menu items. A user will be able to see only the navigation menu items for which he or she has permission to view the relevant objects (instead of the menu items being grayed out).

> That is pretty nice, but I would go further and consider some configuration to hide certain fields from objects, because there is so much information but mos people don't use all of them.

Any discussion along this vein would be wholly out of scope for this FR I'm afraid.
